### DISCUSSION TITLE
Difference between prescribed velocity on material points and on nodes

### DISCUSSION THREAD

**Discussion by PetitPipi** [Anura3D/Anura3D_OpenSource#182] [Q&A] [answered]
Hi Team,

I was working on a model with the prescribed velocity imposed on part of an elastic body to simulate pulling on one end of the body.  With the prescribed velocity on the MP option, when some MPs behind the prescribed velocity MPs (let's call them PVMPs) lost contact with nodes that were connected to the PVMPs (when the PVMP left the cell next to them), the MPs (behind the PVMPs) stopped moving.  This problem didn't occur when I imposed the prescribed velocity on nodes.

![Picture1](https://github.com/Anura3D/Anura3D_OpenSource/assets/106621276/bc73666c-b659-4d92-8ab1-ed68a69d0740)

So I wonder what the difference is between these two options: prescribed velocity on MPs or nodes? 

Best,
P

**Comment by FrancescaCeccato** [comment, marked as answer]
This is a good question. First of all, if you have two bodies, we suggest to use prescribed velocity in combination with moving mesh and eventually contact algorithm.
When you have PV at the nodes of the mesh, all the materials entering the element will get the effect of the prescribed velocity at the node. Therefore it behaves as a boundary codnition at the nodes. If you have contact algorithm, you will always get the same velocity at the contact nodes, not different velocities of the two bodies in contact.
When you have PV at the MPs of a body, this velocity is mapped to the nodes keeping the memory of the entity with prescribed velocity, therefore, If there is contact algorithm, at the nodes of the interace you allow two different velocities and the contact law is enforced.
It is not easy to explain but I hope it is more clear now.

**Reply by PetitPipi** [reply]
Thank you so much!

**Comment by kuailebozai** [comment]
@PetitPipi May I ask you how you use the file loading specified speed, I set the specified speed using the instructions of the ReadPrescribedAccelerationOrVelocityFromFile() program, and created a ".PVF" loading speed file in the calculation folder, but the program can't find this file during the calculation. However, the program cannot find this file during the calculation. I'm not sure if there is an error in my file setup. Have you done this setup before? I was hoping you could guide me in correcting the error. Thank you very much, best wishes.
![微信图片_20240219072455](https://github.com/Anura3D/Anura3D_OpenSource/assets/52518762/31ba2c93-e50a-458f-bb93-6ee6e7c8d8b9)

**Reply by PetitPipi** [reply]
Hello there.  I simply assigned a constant velocity in GiD interface.  I'm sorry. I guess this doesn't help with your problem here.  
I bet you already read the other disscussion about seismic load input in Anura3D, if not, please find it, that might be helpful.  

Best wishes,
P

**Reply by kuailebozai** [reply]
Thank you very much for your reply, I read the discussion about it and I still haven't solved the problem. Best wishes.
